opentracing-api-extensions-tracer/src/main/java/io/opentracing/contrib/api/tracer/APIExtensionsSpan.java

```diff
@@ -230,9 +230,11 @@ public Span setTag(String key, Number value) {
     }
 
     private Span handleSetTag(String key, Object value) {
-        tags.put(key, value);
-        for (SpanObserver observer : observers) {
-            observer.onSetTag(this, key, value);
+        if (key != null && value != null) {
```

Comment:
Are you sure that _both_ must be non-null?
While I understand the `key != null` condition, are you absolutely sure there is not somewhere, someone that has a usecase for tag value `null` ?
(could be, I'm not 100% on the spec)

Reply:
The Java OT API doesn't defined - but definitely key should not be null - and unclear how a null value would be represented, so should probably not be stored. However this change is only impacting whether the observer is informed - it still delegates the supplied values to the underlying tracer to handle how they wish.

One option would be to treat a null value as a remove - but generally tags are only set once.

Reply:
Agreed.

I just wanted to make sure it was a conscious choice, as this library is an 'automatic' wrapper, so might mess things up for wrapped tracers that operate on other assumptions.

I would be weary of limiting input if there's no inherent reason to do so, i.e. if the NPE is not from within the `java-api-extensions` library I would not restrict the input if it is not explicitly forbidden by the spec.

Reply:
Looks like a decent solution, however from a pragmatic point of view maybe double-check whether your original objective is still fixed? 😉 
- https://github.com/opentracing-contrib/java-apache-httpclient/issues/12

I don't know whether the spec allows `null` keys? What would they mean?

Reply:
The tests reproduced the problem - so should be fine.
The spec doesn't make it clear - but atleast now it is delegated to the tracer and observer to handle :)